### DISCUSSION TITLE
Inter-node Communication - encryption

### DISCUSSION THREAD

**Discussion by vshirol** [vernemq/vernemq#1906] [Q&A] [answered]
Is the communication between nodes of the same VerneMQ cluster encrypted? If so please share some details for the same.

**Comment by ioolkos** [comment, marked as answer]
@vshirol 
It is usually not done but possible. There are 2 kinds of connections between 2 cluster nodes. Erlang distribution ones (for exchange of metadata), and MQTT message porting connections. Both can be done. For the latter, you'd have to configure a `vmqs` listener including the options for TLS (see schema file in the `priv` source directory).

Wether encrypted payloads adress the concern might be up for debate. In any case, Verne clusters run behind network proxy/firewall/loadbalancers, not spawning cluster connection over wide area networks or multiple sites.

**Reply by vshirol** [reply]
Thanks @ioolkos. I will take a look on these options. If any further questions on these, will post here.
